### ISSUE TITLE
How to add english as a supported language?

### ISSUE THREAD

**Issue by vantony1** [Tomiinek/Multilingual_Text_to_Speech#18]
Hello,

Thank you for making this code base available! this is absolutely fascinating stuff.

I am working on a research project where I need to produce accented English audio on custom text i.e. I have audios of native speakers which I want to "add an accent" to. I was able to successfully run the two Google Collab notebooks you provided to find that the model is able to output dynamically accented audio. However, I noticed that English is not included in the audio files that the models were trained on.

I want to add English as a supported language. One way to do so that I see is to download the common voice english database or a subset of it, clean it to be in the format of your "cleaned" common voice dataset and then follow the steps to train the models from scratch essentially. There are a couple issues I see with my plan: a) your waveRNN has been pre-trained on CSS10 data which doesn't include english so if thats an issue I might also have to train waveRNN again b) I am not entirely sure yet how to "clean" the common voice data c) the common voice english database is 50GBs which is too big. 

Essentially, I am hoping there is a simpler way to fine-tune the existing models to support English. If you provide me with some direction on this, Id very much appreciate that!

Best,

Vic

**Comment by Tomiinek** [comment]
Hi, thank you, nice to hear :slightly_smiling_face: 

At first, I will ask you a silly question: why do you need a multilingual model? Isn't an multispeaker English text-to-speech model enough for you?  Btw I think I have a monolingual monospeaker English model trained using this code on my local machine. If you want, I can share it.

Your plan is ok, but:

a) It was, it actually does just spectrogram inversion, so it might *somehow* work anyway. You can also use a different model with publicly available pretrained weights (such as WaveGlow).

b-c) You do not need to use Common Voice, English is data rich and has a wide variety of multispeaker datasets. You can for example use LJ Speech and a subset of VCTK to replace CSS10 and CommonVoice. But you should make sure that these English examples are normalized similarly to the rest of the data used for training (i.e, the same punctuation usage, distriution of sentence lenghts and audio durations, ...)

The current code is not very fine-tuning-friendly, unfortunately :cry:

**Comment by vantony1** [comment]
hey! 

Thank you for getting back to me so quickly and for the feedback. I really appreciate it. I will try to go ahead and train the model on normalized data from the suggested datasets.

A part of the research team is working on using multi-speaker English text-to-speech model so I just wanted to explore a different path. That being said, if you could share the monolingual mono-speaker English model trained using this code on your local machine -- I'll go ahead and test it out to see if its produces the kind of results we want it to. And of course, if we end up using either model for the research, we will make sure to give you the appropriate credit in the paper.

Best,

Vic

**Comment by Tomiinek** [comment]
Oh, I see. The english checkpoint is available [here](https://github.com/Tomiinek/Multilingual_Text_to_Speech/releases/download/v1.0/ljspeech_checkpoint.pyt). It produces linear spectrograms (not mel), so you should use the Griffin-Lim algorithm for conversion to audio.

Hope it helps :slightly_smiling_face:
